crates/tests/mqtt_tests/src/test_mqtt_server.rs

```diff
@@ -86,69 +89,76 @@ fn spawn_broker(port: u16) {
     });
 
     std::thread::spawn(move || {
-        let mut rx = tx.connect(200).unwrap();
         tx.subscribe("#").unwrap();
 
         loop {
-            if let Some(message) = rx.recv().unwrap() {
-                for chunk in message.payload.into_iter() {
-                    let mut bytes: Vec<u8> = vec![];
-                    for byte in chunk.into_iter() {
-                        bytes.push(byte);
-                    }
-                    let payload = match std::str::from_utf8(bytes.as_ref()) {
+            let notification = match rx.recv().unwrap() {
+                Some(v) => v,
+                None => continue,
+            };
+            match notification {
+                Notification::Forward(forward) => {
```

Comment:
This code is not easy to read. Why not a simple `while let Some(notification) = rx.recv().unwrap()`?